scripts/stablex/place_margin_orders.js

```diff
@@ -0,0 +1,91 @@
+const BatchExchange = artifacts.require("BatchExchange")
+const { sendTxAndGetReturnValue } = require("../../test/utilities.js")
+const BN = require("bn.js")
+
+const argv = require("yargs")
+  .option("tokens", {
+    describe: "Collection of trusted tokens",
+  })
+  .option("accountId", {
+    describe: "Account index of the order placer",
+  })
+  .option("margin", {
+    type: "float",
+    describe: "Percentage increase required for trade (fees not accounted)",
+    default: 0.2,
+  })
+  .option("sellAmount", {
+    type: "float",
+    describe: "Maximum sell amount (considered infinite if empty)",
+    default: 100,
+  })
+  .option("expiry", {
+    type: "int",
+    describe: "Maximum auction batch for which these orders are valid",
+    default: 2**32 - 1
+  })
+  .demand(["tokens", "accountId"])
+  .help(false)
+  .version(false).argv
+
+
+// TODO - automate this with tokenIdToAddress dot decimals
+const TOKEN_DECMIALS = {
+  2: 6,
+  3: 18,
+  4: 6,
+  5: 18,
+  6: 2,
+  7: 18
+}
+
+module.exports = async callback => {
+  try {
+    const instance = await BatchExchange.deployed()
+    const accounts = await web3.eth.getAccounts()
+    const account = accounts[argv.accountId]
+    console.log(account)
+    const batch_index = (await instance.getCurrentBatchId.call()).toNumber()
+
+    const trusted_tokens = argv.tokens.split(",").map(t => parseInt(t))
+    const expectedReturnFactor = new BN(1 + argv.margin / 100)
+    const sellAmount = new BN(argv.sellAmount)
+    const buyAmount = expectedReturnFactor.mul(sellAmount)
+
+    let buyTokens = []
+    let sellTokens = []
+    let buyAmounts = []
+    let sellAmounts = []
+    for (let i = 0; i < trusted_tokens.length - 1; i++) {
+      const tokenA = trusted_tokens[i]
+      const tokenScaleA = new BN(10).pow(new BN(TOKEN_DECMIALS[tokenA]))
+      for (let j = i + 1; j < trusted_tokens.length; j++) {
+        const tokenB = trusted_tokens[j]
+        const tokenScaleB = new BN(10).pow(new BN(TOKEN_DECMIALS[tokenB]))
+        buyTokens = buyTokens.concat([tokenA, tokenB])
+        sellTokens = sellTokens.concat([tokenB, tokenA])
+        buyAmounts = buyAmounts.concat(
+          buyAmount.mul(tokenScaleA),
+          buyAmount.mul(tokenScaleB)
+        )
+        sellAmounts = sellAmounts.concat(
+          sellAmount.mul(tokenScaleA),
```

Comment:
I suspect this will lead to undesired rounding errors where you can't specify a fractional sell amount. You can do something like this:
```suggestion
          tokenScaleA.muln(argv.sellAmount),
```

The same applies to the other sell amount and the buy amounts.

To prove this works:
```
> `${new BN(1.55).mul(new BN(10000))}`
'10000'
> `${new BN(10000).muln(1.55)}`
'15500'
```

Reply:
What if we insist that `sellAmount` is of type int?

Reply:
Then you wouldn’t be able to sell half a WETH for example. I think it should be easy enough to support fractional sell amounts with the `muln`.